### ISSUE TITLE
Overhaul and update Link accessibility testing

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#7205]
Definition of done: 
- [x] convert existing tests to React Testing Library
- [x] add Accessibility Checker test
- [x] determine and implement missing WCAG 2.1 tests
- [x] determine and document manual testing practices
